src/common/ast-path.js

```diff
@@ -146,6 +146,38 @@ class AstPath {
 
     return true;
   }
+
+  /**
+   * Traverses the current node and its parents (heading toward the tree root)
+   * until it finds a node that matches the provided predicate function. Will
+   * return the current node or the matching ancestor. If no such node exists,
+   * it returns undefined.
```

Comment:
Not sure whether including the current node is useful. I did that only to match the behavior of the DOM method. Let's skip the current node and rename the method (suggestions?), shall we?

Reply:
`closestAncestor`?